packages/utilities/src/dom.ts

```diff
@@ -204,6 +204,31 @@ export function getRect(element: HTMLElement | Window | null): IRectangle | unde
   return rect;
 }
 
+/**
+ * Finds the first parent element where the matchFunctionReturns true
+ * @param element element to start searching at
+ * @param matchFunction the function that determines if the element is a match
+ * @returns the matched element or null no match was found
+ */
+export function findElementRecursive(element: HTMLElement | null, matchFunction: (element: HTMLElement) => boolean): HTMLElement | null {
+  if (!element || element === document.body) {
+    return null;
+  }
+
+  return matchFunction(element) ? element : findElementRecursive(element.parentElement, matchFunction);
+}
+
+/**
+ * Determines if an element, or any of its ancestors, contian the given attribute
+ * @param element - element to start searching at
+ * @param attribute - the attribute to search for
+ * @returns the value of the first instance found
+ */
+export function elementContainsAttribute(element: HTMLElement, attribute: string): string | null {
+  let elementMatch = findElementRecursive(element, element => element.hasAttribute(attribute));
```

Comment:
ERROR: 228:52  typedef               expected arrow-parameter: 'element' to have a typedef
ERROR: 228:52  no-shadowed-variable  Shadowed name: 'element'

Reply:
maybe I can inline fix it for you